ompi/mpi/java/c/mpi_MPI.c

```diff
@@ -1154,6 +1153,9 @@ jboolean ompi_java_exceptionCheck(JNIEnv *env, int rc)
         (*env)->DeleteLocalRef(env, jmessage);
         return JNI_TRUE;
     }
+    else if (JNI_TRUE == jni_exception) {
+        return JNI_TRUE;
```

Comment:
currently, in such a case, the cleanup is performed (e.g. ```ompi_java_releaseWritePtr``` or ```ompi_java_releaseReadPtr```) but a ```java.lang.*Exception``` is thrown instead of a ```mpi.MPIException```, am i right ?

what if there is an MPI error (for example, root rank is negative ) ? in my understanding, no cleanup is performed.

at first, could/should we
- do the cleanup first
- and then ```ompi_java_exceptionCheck``` ?

an other way to put things is, is there any reason __not__ to do any cleanup if the MPI subroutine returned an error ?

if the buffer is too short, throwing a ```mpi.MPIException``` is still half baked to me, since blocking MPI subroutines would throw a ```mpi.MPI_Exception``` but non blocking MPI subroutines would throw a ```java.lang.*Exception``` and likely after the MPI subroutine returned successfully.

bottom line, i do not feel comfortable with that kind of approach.
- having Java throw a ```java.lang.*Exception``` is a valid option to me
- having MPI Java bindings check user buffer (a la ```memchecker```) when directed by the user (this is not cheap from a performance point of view) can be seen as an interesting but __new__ feature